apps/etterlatte-saksbehandling-ui/client/src/components/behandling/sanksjon/Sanksjon.tsx

```diff
@@ -0,0 +1,272 @@
+import styled from 'styled-components'
+import { useApiCall } from '~shared/hooks/useApiCall'
+import React, { useEffect, useState } from 'react'
+import Spinner from '~shared/Spinner'
+import { ApiErrorAlert } from '~ErrorBoundary'
+import { IBehandlingReducer } from '~store/reducers/BehandlingReducer'
+import { behandlingErRedigerbar } from '~components/behandling/felles/utils'
+import { isPending, mapApiResult } from '~shared/api/apiUtils'
+import { useInnloggetSaksbehandler } from '../useInnloggetSaksbehandler'
+import { BodyShort, Button, Detail, Heading, HStack, Table, TextField, VStack } from '@navikt/ds-react'
+import { PencilIcon } from '@navikt/aksel-icons'
+import { formaterStringDato } from '~utils/formattering'
+import { ControlledMaanedVelger } from '~shared/components/maanedVelger/ControlledMaanedVelger'
+import { useForm } from 'react-hook-form'
+import { startOfDay, formatISO } from 'date-fns'
+import { hentSanksjon, lagreSanksjon } from '~shared/api/sanksjon'
+import { TableWrapper } from '~components/behandling/beregne/OmstillingsstoenadSammendrag'
+
+export interface ISanksjon {
+  id?: string
+  behandlingId: string
+  sakId: number
+  fom: string
+  tom?: string
+  opprettet: {
+    tidspunkt: string
+    ident: string
+  }
+  endret?: {
+    tidspunkt: string
+    ident: string
+  }
+  beskrivelse: string
+}
+
+export interface ISanksjonLagre {
+  id?: string
+  sakId: number
+  fom: string
+  tom?: string
+  beskrivelse: string
+}
+
+export const Sanksjon = ({ behandling }: { behandling: IBehandlingReducer }) => {
+  const [lagreSanksjonResponse, lagreSanksjonRequest] = useApiCall(lagreSanksjon)
+  const [sanksjonStatus, hentSanksjonRequest] = useApiCall(hentSanksjon)
+  const [sanksjoner, setSanksjoner] = useState<ISanksjon[]>()
+  const [visForm, setVisForm] = useState(false)
+  const innloggetSaksbehandler = useInnloggetSaksbehandler()
+
+  const redigerbar = behandlingErRedigerbar(
+    behandling.status,
+    behandling.sakEnhetId,
+    innloggetSaksbehandler.skriveEnheter
+  )
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    getValues,
+    reset,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      datoFom: undefined,
+      datoTom: undefined,
+      beskrivelse: '',
+    },
+  })
+
+  const validerFom = (value: Date): string | undefined => {
+    const fom = startOfDay(new Date(value))
+    const tom = startOfDay(new Date(getValues().datoTom!))
```

Comment:
Det føles litt hacky å bruke en invalid date når tom ikke er satt -- det blir riktig nok i valideringen men det er ikke åpenbart hvorfor (og så kan man heller ikke flippe på noen av sammenligningene uten at ting knekker, som er ganske uventet hvis man ikke graver i oppførselen).

Jeg tenker det kanskje blir lettere å følge koden / mindre skjørt hvis man flytter valideringen til skjema-nivå, og ikke per-felt-nivå? Så kan submitSanksjon sjekke at ting stemmer før den sender av gårde

Reply:
Har flyttet validering til submitSanksjon og forenklet logikken for lettere å skjønne valideringen. Kan potensielt flytte validering til egen funksjon, men vet ikke helt om det gir noe merverdi.